### DISCUSSION TITLE
[Suggestion] "Superstructures"

### DISCUSSION THREAD

**Discussion by AlienXtream** [tsunamayo/Starship-EVO#4204] [Suggestions]
this is an idea i had while replying to another thread on here. basically a system to enable MASSIVE scale objects like the deathstar. at its core the "superstructure" base would basically be no different from how the ring world is handled. rather than being a dynamic object, it would be fixed in space (or on a predefined orbit when that is implemented) and basically have no physics other than collision. the purpose of this object would be to serve as a "terrain" for buildings to be built onto. the form and shape of these superstructures would probably be fixed (unless you want to make them with the same method you end up using for planets terrain destruction, assuming you go with that approach). as such they would be prefabbed and have a basic form to build upon at the "small" scale but detailed enough so that the LoD looks good. in survival these structures would not be cheap and get more expensive with size. to build one the player would have to go through a few hoops. first would be getting permission to build such a massive permanent structure, thus requiring significant influence with the faction in control of that sector. influence could be anything from notoriety to diplomatic seniority, basically whatever system that gets made to determine if the faction is hostile or friendly or whatever. point is that you would need to convince them to let you build this thing. the next thing you would need is the  "currency" to afford an "orbital lane". once you have an orbital lane you would then begin construction. unlike the ship/base building present elsewhere in the game however, this would be more of a "resource sink" project. the shape is defined and the cost scaled depending on the size you want. you then provide the "orbital construction crew" (some sort of temporary space station close to the site) resources to start the core. the core is basically the foundation. once tier one is done tier 2 gets under way. as you give more and more resources to the construction crew more and more of the structure will form (with some time between stages or even some sort of animation as the crew build). the final stage would be much cheaper and basically be where you chose the outward appearance of the structure (material, color, layout of the "openings"[from a list of options] etc). once done the structure is permanent unless the core is destroyed (think death star trench run in star wars) or the save/server is wiped. so what is the point of this? well, other than being cool, the core could be used to store the positions and ID's of blueprints and because of the inherently larger size they could serve quite nicely as the basis for large scale dungeons comprised of several independent pieces. think abandon facilities or pirate outposts. additionally the same system could be adapted for players to "claim" portions of planets and build a town or spaceport. 

mind you none of this is trying to say "make this" or "do that". its your game at the end of the day. i just hope that you get some good ideas from it that you can make use of :)

im sorry this sort of started to get long winded. its 10pm for me and i am practically falling asleep but i wanted to share the idea before i forget about it.

**Comment by Hecateus** [comment]
Tsunamayo has mentioned aiming to expand on superstructures in the future, but he seems to on finishing the pre-game loop stuff at this point in time.  Game-ifying of megastructures will be a long way off.  But for spitballing here is some inspiration: https://www.youtube.com/watch?v=HlmKejRSVd8&list=PLIIOUpOge0LtW77TNvgrWWu5OC3EOwqxQ

**Comment by WILDCATreactor** [comment]
as far as "currency" megastructures would need to leverage massive strength-to-weight ratios, so setting up, let's say... "buckytube braiding" production chains (i.e. carbon nanotube farms > molecular spindles > buckytube looms) creates a set of block tiers (T2 nanocarbon blocks, bricks, and armor > T2.5 carbonised superconductors (for reactors/shields/weapons > T3 buckytube fiber blocks, bricks, armor, and "M.Structure Currency") that builds into producing the Tier 3 megastructure mats, so even if you haven't reached the end of the production chain or can't afford T3 mass production, you're getting useful items out of it.

As far as having Faction Reputation or whatever, because spitballing systems is fun, what if... like, each planet had a certain number of orbital lanes, each lane occupied by a station or superstructure doubled the Reputation cost of the next one to be claimed by a charter or whatever and each station or superstructure or whatever a player/faction had also doubled the Reputation cost of claiming additional lanes?  So, the first Player Orbital Lane claimed was at 1x Rep, the second is 2x Rep -OR- if claimed by the same player/faction would be 4x Rep; that same player putting a second structure in another system controlled by the same NPC faction would only have to pay 2x Rep (if they were the first players to put a structure in that system), representing NPCs being unwilling to allow a potential competitor or many competitors to build up power base(s) under their nose.  If Reputation is on a finite scale (say, up to 100), then players and player factions can only put so many structures in a given territory, regardless of whether they spread everything across multiple systems or not.

Now, if a Player/Player Faction conquers a star system, they CREATE their Reputation as a faction by building structures in a controlled system, and other players/factions can use Rep they accumulate by trading or doing jobs with that Player/Faction to put structures inside of the player-controlled system.  So, the more heavily built-up a Player-controlled star system is, the more other players and factions can establish themselves within that system as it's settled and exploited.

- Reputation gained with NPC Factions can be spent or "occupied" by creating structures in their territory.
- Reputation has a maximum value, ultimately capping how many structures a player/faction can place in one NPC's territory.
- Removing a structure refunds some or all of the "spent" Reputation.
- As a star system is increasingly settled, Rep costs increase geometrically.
- Rep costs per additional structure also increase geometrically for each player/faction, preventing one player/group from making large landgrabs.
- Player-controlled territories gain Reputation as the player/faction place more structures.
- Other players can spend Rep with controlling players/factions to place structures in their territory, up to the max Rep in each star system with the same geometric cost increases as placing structures in NPC territories.

- Stations cost an amount of Rep equal to their volume factor
- Multiple stations can occupy different sections of the same "Orbital Lane" around a planet or moon
- A single Superstructure, such as a Ringworld, takes up an entire Orbital Lane and costs a large amount of Rep to "charter" that orbital lane.
- Structures built by a controlling Player/Faction create Rep that others can use to place stations or charter orbital lanes in the same way.
- Superstructures cost an enormous amount of materials, including Tier 3 supermaterials required for the load-bearing elements.
- Tier 3 supermaterials require a modest production chain to be established and done so en masse, established industry is critical both to build the structure as well as serve the fleet that will be needed to defend it against interlopers.
Anything I might've forgotten to consider?


As far as types of superstructures?  Most of them are planet-sized, Halo-style ringworlds (the ring in STEVO is actually fairly small comparatively, I just had to check on the Halos, and holy crap, diameter of Earth), mobile planetoids, and maybe O'Neill cylinders are the only ones that seem reasonable in scale for the game's purposes.

**Comment by tsunamayo** [comment]
I havent read 100% of this but yeah I thought about something along those line already. Sky is the limit, but this is a long term goal of course. Thanks

**Comment by AlienXtream** [comment]
oh yeh its long term XD. would probably best use the same method as planets do. similar things can be done in space engineers with terrain voxels. would basically be a "planet" with an arbitrary shape rather than a spheroid. but indeed its a long term thing. even the death star took over a decade to build and that is a fictional space station.

**Comment by WILDCATreactor** [comment]
like a literal sandbox? big grey megastructure that holds a landscape section or multiple sections with small walls in between (i.e. Halo's Ark) each landscape section can be sculpted in an editor tool/mode or whatever so the owner can have custom terrain, set the water level, etc, and then save the changes and upload them into the actual game object
